### PR TITLE
Unify `error=` keyword

### LINKED ISSUE #135: Change function signature of `is_` functions to be consistent with `isapprox`.
To be consistent with the newly planned error mode, these two positional arguments of `is_point` (for the first) and `is_vector` these two should become keyword arguments.
This should also be done for the internal `check_` functions also for consistency with the `isapprox` and `check_approx` functions,
see also discussions in #134.

### PR DESCRIPTION
**Pull request by kellertuer**
This resolves #135.

### CODE REVIEW

**Comment by kellertuer** [comment]
But there seem to be a few dispatches through the decorators that are quite tricky still.

**Comment by kellertuer** [comment]
One thing missing as well is that `:info` and `:warn` were previously only used on DomainErrors, now that they here also end up being used for ManifoldDomainErrors, the string generation fails. This I have to take a look at what best to do.

**Comment by kellertuer** [comment]
Hmpf.

In principle this PR is finished – Docs are updated (but the push to 1.0 comes from another PR, the tutorial is updated here already)
But also vor now downloading packages on CI errors with a 502, so we have to wait for the CodeCov a bit. Still, this 3rd PR should finish the collection of breaking changes for 0.15.0

**Comment by kellertuer** [comment]
I just had to adapt 2 tests from the other PRs, merged the NEWS – there we only have to enter the date when we merge/register 0.15

File: src/ValidationManifold.jl [hunk 1]
```diff
@@ -8,9 +8,20 @@ encapsulated/stripped automatically when needed.
 
 This manifold is a decorator for a manifold, i.e. it decorates a [`AbstractManifold`](@ref) `M`
 with types points, vectors, and covectors.
+
+# Constructor
+
+ValidationManifold(M; error::Symbol = :error)
```
Comment (mateuszbaran) [inline src/ValidationManifold.jl:14]:
```suggestion
    ValidationManifold(M::AbstractManifold; error::Symbol = :error)
```

**Comment by kellertuer** [comment]
Yes. I wrote that in detail in the NEWS.md, to unify all signatures I had to switch 2 positionals.

**Comment by mateuszbaran** [comment]
I see, sorry -- I forgot to read NEWS.md.

**Comment by mateuszbaran** [comment]
The CI issue is really werid...

**Comment by mateuszbaran** [comment]
I've tried tracking it and I suspect https://github.com/JuliaRegistries/General/pull/93498 is the issue.

**Comment by mateuszbaran** [comment]
I've mostly updated Manifolds.jl. Could you take a look why a wrong exception type is throws in `centered_matrices.jl`, `fixed_rank.jl` and `probability_simplex.jl`?

**Comment by kellertuer** [comment]
Not today, maybe tomorrow before my lecture, otherwise only early evening (travelling to Oslo in the afternoon)

**Comment by mateuszbaran** [comment]
Thanks!

**Pull request merged**